recipes/folly/all/conanfile.py

```diff
@@ -181,6 +184,7 @@ def generate(self):
                 tc.variables["CMAKE_C_FLAGS"] = "/arch:FMA"
                 tc.variables["CMAKE_CXX_FLAGS"] = "/arch:FMA"
 
+        tc.variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

Comment:
Should be set by the toolchain... Folly must be messy with this 😂

Reply:
What I have seen is that when `folly:shared=True` then `conan_toolchain.cmake` doesn't define CMAKE_POSITION_INDEPENDENT_CODE which yields this compilation error:

https://c3i.jfrog.io/c3i/misc/logs/pr/15726/2-linux-gcc/folly/2019.10.21.00//7ed9e0b4a97c77d17a821f8f65ec95e88ef5dee0-build.txt

Reply:
```suggestion
        # Folly is not respecting this from the helper https://github.com/conan-io/conan-center-index/pull/15726/files#r1097068754
        tc.variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```